patches/minecraft/net/minecraft/server/players/PlayerList.java.patch

```diff
@@ -24,9 +24,19 @@
        servergamepacketlistenerimpl.m_141995_(new ClientboundUpdateRecipesPacket(this.f_11195_.m_129894_().m_44051_()));
        servergamepacketlistenerimpl.m_141995_(new ClientboundUpdateTagsPacket(this.f_11195_.m_129895_().m_144442_(this.f_11206_)));
        this.m_11289_(p_11263_);
-@@ -176,7 +_,7 @@
- 
-       this.m_11264_(mutablecomponent.m_130940_(ChatFormatting.YELLOW), ChatType.SYSTEM, Util.f_137441_);
+@@ -167,16 +_,9 @@
+       p_11263_.m_8952_().m_12789_(p_11263_);
+       this.m_11273_(serverlevel1.m_6188_(), p_11263_);
+       this.f_11195_.m_129929_();
+-      MutableComponent mutablecomponent;
+-      if (p_11263_.m_36316_().getName().equalsIgnoreCase(s)) {
+-         mutablecomponent = new TranslatableComponent("multiplayer.player.joined", p_11263_.m_5446_());
+-      } else {
+-         mutablecomponent = new TranslatableComponent("multiplayer.player.joined.renamed", p_11263_.m_5446_(), s);
+-      }
+-
+-      this.m_11264_(mutablecomponent.m_130940_(ChatFormatting.YELLOW), ChatType.SYSTEM, Util.f_137441_);
++      net.minecraftforge.common.ForgeHooks.onJoinMessageFormat(p_11263_,this, s);
```

Comment:
Same here, forge hook is responsible for providing the message, not sending it as well. Also, this could be made into a much smaller patch by just checking if the mutable component is not null and then doing an else if.